### PR TITLE
refactor(core): convert applicationCtrl to TS, executions to component

### PR DESCRIPTION
**Pull request by anotherchrisberry**
I considered converting these to React but...not yet. For now, just converting to Typescript.

The application conversion is to avoid the bad layout issue on initial load on a deep link to a server group/instance/load balancer. We'll now hide the details panel until the application is ready, which prevents the shifting of it to the right when the filters are ready.

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
@jrsquared or @christopherthielen PTAL

File: app/scripts/modules/core/src/application/application.component.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+import './newapplication.less';
+import './application.less';
+
+export class ApplicationController implements IComponentController {
+
+  public app: Application;
+  public refreshTooltipTemplate = require('./applicationRefresh.tooltip.html');
+
+  constructor(private recentHistoryService: RecentHistoryService, private $uibModal: IModalService) {}
```
Comment (jrsquared) [inline app/scripts/modules/core/src/application/application.component.ts]: `'ngImport';`

File: app/scripts/modules/core/src/insight/insightLayout.component.ts [hunk 2]
```diff
@@ -1,8 +1,15 @@
 import {InsightFilterStateModel} from './insightFilterState.model';
 import {INSIGHT_NGMODULE} from './insight.module';
+import { Application } from 'core/application';
 
 class InsightLayoutCtrl {
+  public app: Application;
+  public ready = false;
   constructor(public InsightFilterStateModel: InsightFilterStateModel) { 'ngInject'; }
```
Comment (jrsquared) [inline app/scripts/modules/core/src/insight/insightLayout.component.ts]: space

File: app/scripts/modules/core/src/application/application.controller.spec.js [hunk 3]
```diff
@@ -1,30 +0,0 @@
-'use strict';
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/application/application.controller.spec.js]: Why did you delete this beautiful test?

Comment (anotherchrisberry) [inline app/scripts/modules/core/src/application/application.controller.spec.js]: it felt awful to remove it but the damage is done, can't undelete tests

File: app/scripts/modules/core/src/application/application.less [hunk 4]
```diff
@@ -1,5 +1,11 @@
 @import (reference) "~core/presentation/less/imports/commonImports.less";
 
+application {
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/application/application.less]: We will need to migrate this to a class when we switch to react.  Since this is new styling, maybe we should use a class now

File: app/scripts/modules/core/src/application/application.state.provider.ts [hunk 5]
```diff
@@ -99,9 +99,7 @@ export class ApplicationStateProvider implements IServiceProvider {
     };
     applicationConfig.views = {};
     applicationConfig.views[mainView] = {
-      templateUrl: require('../application/application.html'),
-      controller: 'ApplicationCtrl',
-      controllerAs: 'ctrl'
+      template: `<application app="$resolve.app"></application>`,
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/application/application.state.provider.ts]: Now that we're on ui-router 1.0+ you can write:
```
component: 'application',
```

Comment (christopherthielen) [inline app/scripts/modules/core/src/application/application.state.provider.ts]: (although there's nothing wrong with this template string)

File: app/scripts/modules/core/src/delivery/delivery.states.ts [hunk 6]
```diff
@@ -72,7 +72,7 @@ module(DELIVERY_STATES, [
     url: `?${stateConfigProvider.paramsToQuery(filterModelConfig)}`,
     views: {
       'pipelines': {
-        template: '<executions application="application"></executions>',
+        template: '<executions application="$resolve.app"></executions>',
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/delivery/delivery.states.ts]: Again, this is fine.  There are a few ways to map the resolve to the component input in 1.0 way:

```
pipelines': {
  component: 'executions',
  bindings: { application: 'app' },
}
```
https://ui-router.github.io/ng1/docs/latest/interfaces/ng1.ng1viewdeclaration.html#bindings

Or, change the `executions` input binding to `app` to get the implicit mapping
```
const executionsComponent: IComponentOptions = {
   bindings: {
     application: '<app',
   },
```

File: app/scripts/modules/core/src/insight/insightLayout.component.ts [hunk 7]
```diff
@@ -1,8 +1,15 @@
...
+
+  public $onInit() {
+    this.app.ready().then(() => this.ready = true);
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/insight/insightLayout.component.ts]: a lot of this state management stuff could be moved into the router/state machine

**Comment by christopherthielen** [comment]
It's already purple, but LGTM

File: app/scripts/modules/core/src/application/application.less (hunk 4 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/application/application.less]: A class would work, but we'd have to put it on the component itself, i.e. `<application class="application"/>`. I bet we'll remember (because nothing will be scrollable on the page).

File: app/scripts/modules/core/src/application/application.state.provider.ts (hunk 5 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/application/application.state.provider.ts]: I like that better, will update in a separate PR.
